File: retaguarda-software-house/c-sharp/T2TiRetaguardaSH/T2TiRetaguardaSH/Models/Transiente/RetornoJsonErro.cs
using System;

namespace T2TiRetaguardaSH.Models
{
    public class RetornoJsonErro
    {
        public RetornoJsonErro(int codigo, String mensagem, Exception ex)
        {
            Status = codigo;
            Message = mensagem;

            switch (Status)
            {
                case 400:
                    Error = "Bad Request";
                    break;
                case 404:
                    Error = "Not Found";
                    break;
                case 500:
                    Error = "Internal Server Error";
                    break;
                default:
                    Error = "Erro não mapeado";
                    break;
            }

            if (ex != null)
            {
                Message = Message + " - Exceção: " + ex.Message;
                Trace = ex.StackTrace;
            }

        }


        public int Status { get; set; }
        public string Error { get; set; }
        public string Message { get; set; }
        public string Trace { get; set; }

    }
}

File: projetos-fenix-para-estudo/t2ti-erp-fenix-servidores-provider-inicio/fontes/backend/php/fenix/src/model/transiente/RetornoJsonErro.php
<?php

/// classe transiente que armazena o objeto do erro que será retornado para o cliente
class RetornoJsonErro implements \JsonSerializable
{
    private $status;
    private $message;
    private $error;
    private $trace;

    public function __construct(int $codigo, string $mensagem, ?Exception $erro)
    {
        $this->status = $codigo;
        $this->message = $mensagem;

        switch ($this->status) {
            case 400:
                $this->error = 'Bad Request';
                break;
            case 404:
                $this->error = 'Not Found';
                break;
            case 500:
                $this->error = 'Internal Server Error';
                break;
            default:
                $this->error = 'Erro não mapeado';
                break;
        }

        if ($erro != null) {
            $this->message = $this->message . " - Exceção: " . $erro->getMessage();
            $this->trace = $erro->getTraceAsString();
        }
    }

    /**
     * {@inheritdoc}
     */
    public function jsonSerialize()
    {
        return [
            'status' => $this->status,
            'message' => $this->message,
            'error' => $this->error,
            'trace' => $this->trace
        ];
    }


}